phf_shared/src/lib.rs

```diff
@@ -103,6 +106,14 @@ impl PhfHash for [u8] {
     }
 }
 
+#[cfg(feature = "unicase")]
+impl<S> PhfHash for unicase::UniCase<S>
+where unicase::UniCase<S>: Hash {
+    #[inline]
+    fn phf_hash<H: Hasher>(&self, state: &mut H) {
+        self.hash(state)
```

Comment:
Does UniCase's Hash impl depend on the size of `usize`? I seem to remember that `str`'s implementation writes its length first, for example. This'll break things when cross compiling.